### PR TITLE
Release Zebra 1.0.0-rc.0

### LINKED ISSUE #3050: Tag Zebra 1.0.0-rc.0 stable release candidate
Once all of the items in #3096 are completed we will tag our first Zebra 1.0.0 stable release to include all of the NU5 (and prior) consensus rule validation and ensure that Zebra actively contributes to the Zcash network by relaying transactions and blocks.

### PR DESCRIPTION
**Pull request by arya2**
Close #3050

### How to Increment Versions

Zebra follows [semantic versioning](https://semver.org).
Semantic versions look like: MAJOR`.`MINOR`.`PATCH[`-`TAG`.`PRE-RELEASE]

The [draft `zebrad` changelog](https://github.com/ZcashFoundation/zebra/releases) will have an automatic version bump. This version is based on [the labels on the PRs in the release](https://github.com/ZcashFoundation/zebra/blob/main/.github/release-drafter.yml).

Check that the automatic `zebrad` version increment is correct:

If we're releasing a mainnet network upgrade, it is a `major` release:
1. Increment the `major` version of _*all*_ the Zebra crates.
2. Increment the `patch` version of the tower crates.

If we're not releasing a mainnet network upgrade, check for features, major changes, deprecations, and removals. If this release has any, it is a `minor` release:
1. Increment the `minor` version of `zebrad`.
2. Increment the `pre-release` version of the other crates.
3. Increment the `patch` version of the tower crates.

Otherwise, it is a `patch` release:
1. Increment the `patch` version of `zebrad`.
2. Increment the `pre-release` version of the other crates.
3. Increment the `patch` version of the tower crates.

Zebra's Rust API is not stable or supported, so we keep all the crates on the same beta `pre-release` version.

### Version Locations

Once you know which versions you want to increment, you can find them in the:

zebrad (rc):
- [x] zebrad `Cargo.toml`
- [x] `zebra-network` protocol user agent: https://github.com/ZcashFoundation/zebra/blob/main/zebra-network/src/constants.rs
- [x] `README.md`
- [x] `book/src/user/install.md`

crates (pre-release):
- [x] zebra-* `Cargo.toml`s

tower (patch):
- [x] tower-* `Cargo.toml`s

auto-generated:
- [x] `Cargo.lock`: run `cargo build` after updating all the `Cargo.toml`s

#### Version Tooling

You can use `fastmod` to interactively find and replace versions.

For example, you can do something like:
```
fastmod --extensions rs,toml,md --fixed-strings '1.0.0-rc.0' '1.0.0-rc.1' zebrad README.md book/src/user/install.md zebra-network/src/constants.rs
fastmod --extensions rs,toml,md --fixed-strings '1.0.0-beta.15' '1.0.0-beta.16' zebra-*
fastmod --extensions rs,toml,md --fixed-strings '0.2.30' '0.2.31' tower-batch tower-fallback
```

If you use `fastmod`, don't update versions in `CHANGELOG.md`.

## README

Update the README to:
- [x] Remove any "Known Issues" that have been fixed
- [x] Update the "Build and Run Instructions" with any new dependencies.
      Check for changes in the `Dockerfile` since the last tag: `git diff <previous-release-tag> docker/Dockerfile`.

## Checkpoints

With every release and for performance reasons, we want to update the zebra checkpoints. More information on how to do this can be found in [the zebra-checkpoints README](https://github.com/ZcashFoundation/zebra/blob/main/zebra-consensus/src/checkpoint/README.md).

To do this you will need a synchronized `zcashd` node. You can request help from other zebra team members to submit this PR if you can't make it yourself at the moment of the release.

## Change Log

**Important**: Any merge into `main` deletes any edits to the draft changelog.
Once you are ready to tag a release, copy the draft changelog into `CHANGELOG.md`.

We use [the Release Drafter workflow](https://github.com/marketplace/actions/release-drafter) to automatically create a [draft changelog](https://github.com/ZcashFoundation/zebra/releases).

We follow the [Keep a Changelog](https://keepachangelog.com/en/1.0.0/) format.

To create the final change log:
- [x] Copy the draft changelog into `CHANGELOG.md`
- [x] Delete any trivial changes. Keep the list of those, to include in the PR
- [x] Combine duplicate changes
- [x] Edit change descriptions so they are consistent, and make sense to non-developers
- [x] Check the category for each change
  - Prefer the "Fix" category if you're not sure

<details>

#### Change Categories

From "Keep a Changelog":
* `Added` for new features.
* `Changed` for changes in existing functionality.
* `Deprecated` for soon-to-be removed features.
* `Removed` for now removed features.
* `Fixed` for any bug fixes.
* `Security` in case of vulnerabilities.

</details>

### Create the Release PR

After you have the version increments, the updated checkpoints and the updated changelog:

- [x] Make sure the PR with the new checkpoint hashes is already merged, or make it part of the changelog PR
- [x] Push the version increments and the updated changelog into a branch
      (name suggestion, example: `v100-rc0-release`)
- [x] Create a release PR by adding `&template=release-checklist.md` to the
      comparing url ([Example](https://github.com/ZcashFoundation/zebra/compare/v100-alpha0-release?expand=1&template=release-checklist.md)).
  - [x] Add the list of deleted changelog entries as a comment to make reviewing easier.
- [x] Turn on [Merge Freeze](https://www.mergefreeze.com/installations/3676/branches).
- [x] Once the PR is ready to be merged, unfreeze it [here](https://www.mergefreeze.com/installations/3676/branches).
      Do not unfreeze the whole repository.

### Create the Release

- [x] Once the PR has been merged, create a new release using the draft release as a base,
      by clicking the Edit icon in the [draft release](https://github.com/ZcashFoundation/zebra/releases)
- [x] Set the tag name to the version tag,
      for example: `v1.0.0-rc.0`
- [x] Set the release to target the `main` branch
- [x] Set the release title to `Zebra ` followed by the version tag,
      for example: `Zebra 1.0.0-rc.0`
- [x] Replace the prepopulated draft changelog in the release description by the final
      changelog you created; starting just _after_ the title `## [Zebra ...` of
      the current version being released, and ending just _before_ the title of
      the previous release.
- [x] Mark the release as 'pre-release', until it has been built and tested
- [x] Publish the pre-release to GitHub using "Publish Release"

## Binary Testing

- [x] Wait until the [Docker binaries have been built on `main`](https://github.com/ZcashFoundation/zebra/actions/workflows/continous-integration-docker.yml), and the quick tests have passed.
      (You can ignore the full sync and `lightwalletd` tests, because they take about a day to run.)
- [x] [Publish the release to GitHub](https://github.com/ZcashFoundation/zebra/releases) by disabling 'pre-release', then clicking "Publish Release"
- [x] Wait until [the Docker images have been published](https://github.com/ZcashFoundation/zebra/actions/workflows/release-binaries.yml)
- [x] Test the Docker image using `docker run zfnd/zebra:1.0.0-rc.<version>` 
- [x] Turn off [Merge Freeze](https://www.mergefreeze.com/installations/3676/branches) for the whole repository

If building or running fails after tagging:
1. Fix the bug that caused the failure
2. Increment versions again, following these instructions from the start
3. Update the code and documentation with a **new** git tag
4. Update `CHANGELOG.md` with details about the fix
5. Tag a **new** release

### CODE REVIEW

**Comment by teor2345** [comment]
`fake_peer_set::mempool_requests_for_transactions` failed with:
> internal error: entered unreachable code: `MempoolTransactionIds` requests should always respond `Ok(Vec<UnminedTxId>)`, got Ok(Nil)

https://github.com/ZcashFoundation/zebra/actions/runs/3229701831/jobs/5287294141#step:14:9950

**Comment by teor2345** [comment]
(This isn't a release blocker.)

**Comment by teor2345** [comment]
Removed changelog entries:
- build(deps): Bump metrics and metrics-exporter-prometheus ([#5256](https://github.com/ZcashFoundation/zebra/pull/5256))
- build(deps): bump actions/checkout from 3.0.2 to 3.1.0 ([#5329](https://github.com/ZcashFoundation/zebra/pull/5329))
- build(deps): bump actions/github-script from 6.2.0 to 6.3.0 ([#5271](https://github.com/ZcashFoundation/zebra/pull/5271))
- build(deps): bump actions/github-script from 6.3.0 to 6.3.1 ([#5317](https://github.com/ZcashFoundation/zebra/pull/5317))
- build(deps): bump codecov/codecov-action from 3.1.0 to 3.1.1 ([#5205](https://github.com/ZcashFoundation/zebra/pull/5205))
- build(deps): bump color-eyre from 0.6.1 to 0.6.2 ([#5220](https://github.com/ZcashFoundation/zebra/pull/5220))
- build(deps): bump inferno from 0.11.7 to 0.11.8 ([#5290](https://github.com/ZcashFoundation/zebra/pull/5290))
- build(deps): bump inferno from 0.11.8 to 0.11.9 ([#5306](https://github.com/ZcashFoundation/zebra/pull/5306))
- build(deps): bump insta from 1.19.1 to 1.20.0 ([#5198](https://github.com/ZcashFoundation/zebra/pull/5198))
- build(deps): bump insta from 1.20.0 to 1.21.0 ([#5291](https://github.com/ZcashFoundation/zebra/pull/5291))
- build(deps): bump itertools from 0.10.4 to 0.10.5 ([#5209](https://github.com/ZcashFoundation/zebra/pull/5209))
- build(deps): bump once\_cell from 1.14.0 to 1.15.0 ([#5216](https://github.com/ZcashFoundation/zebra/pull/5216))
- build(deps): bump reqwest from 0.11.11 to 0.11.12 ([#5215](https://github.com/ZcashFoundation/zebra/pull/5215))
- build(deps): bump reviewdog/action-actionlint from 1.29.0 to 1.30.0 ([#5195](https://github.com/ZcashFoundation/zebra/pull/5195))
- build(deps): bump reviewdog/action-actionlint from 1.30.0 to 1.31.0 ([#5238](https://github.com/ZcashFoundation/zebra/pull/5238))
- build(deps): bump reviewdog/action-actionlint from 1.31.0 to 1.32.0 ([#5318](https://github.com/ZcashFoundation/zebra/pull/5318))
- build(deps): bump reviewdog/action-actionlint from 1.32.0 to 1.33.0 ([#5368](https://github.com/ZcashFoundation/zebra/pull/5368))
- build(deps): bump ripemd from 0.1.1 to 0.1.3 ([#5248](https://github.com/ZcashFoundation/zebra/pull/5248))
- build(deps): bump serde from 1.0.144 to 1.0.145 ([#5247](https://github.com/ZcashFoundation/zebra/pull/5247))
- build(deps): bump thiserror from 1.0.35 to 1.0.37 ([#5285](https://github.com/ZcashFoundation/zebra/pull/5285))
- build(deps): bump tj-actions/changed-files from 29.0.7 to 31.0.1 ([#5254](https://github.com/ZcashFoundation/zebra/pull/5254))
- build(deps): bump tj-actions/changed-files from 29.0.7 to 32.0.0 ([#5354](https://github.com/ZcashFoundation/zebra/pull/5354))
- build(deps): bump tokio from 1.21.1 to 1.21.2 ([#5289](https://github.com/ZcashFoundation/zebra/pull/5289))
- build(deps): bump tokio-stream from 0.1.9 to 0.1.10 ([#5197](https://github.com/ZcashFoundation/zebra/pull/5197))
- build(deps): bump tracing from 0.1.31 to 0.1.35 ([#5225](https://github.com/ZcashFoundation/zebra/pull/5225))
- build(deps): bump tracing from 0.1.35 to 0.1.36 ([#5273](https://github.com/ZcashFoundation/zebra/pull/5273))
- build(deps): bump tracing-subscriber from 0.3.11 to 0.3.15 ([#5223](https://github.com/ZcashFoundation/zebra/pull/5223))
- build(deps): bump uint from 0.9.3 to 0.9.4 ([#5208](https://github.com/ZcashFoundation/zebra/pull/5208))
- build(deps): bump w9jds/firebase-action from 11.5.0 to 11.9.0 ([#5196](https://github.com/ZcashFoundation/zebra/pull/5196))
- change(ci): Combine high and low Mergify queues into a batched queue ([#5201](https://github.com/ZcashFoundation/zebra/pull/5201))
- change(ci): Create a new cached state image every 12 hours ([#5191](https://github.com/ZcashFoundation/zebra/pull/5191))
- change(dev): Add CHANGELOG checks to PR review template ([#5202](https://github.com/ZcashFoundation/zebra/pull/5202))
- change(doc): Require Rust 1.63 in the README ([#5359](https://github.com/ZcashFoundation/zebra/pull/5359))
- change(release): Automatically split CHANGELOG entries into categories ([#5203](https://github.com/ZcashFoundation/zebra/pull/5203))
- ci(deploy): retry `gcloud` ssh connections if it fails ([#5292](https://github.com/ZcashFoundation/zebra/pull/5292))
- ci(lint): do not fail linter on `push` events ([#5331](https://github.com/ZcashFoundation/zebra/pull/5331))
- ci(ssh): connect using `ssh-compute` action by Google ([#5330](https://github.com/ZcashFoundation/zebra/pull/5330))
- ci(ssh): revert using `ssh-compute` action \& increase sshd connection limit ([#5367](https://github.com/ZcashFoundation/zebra/pull/5367))
- doc(ci): Explain how to fix Mergify and Zcash parameter download failures ([#5240](https://github.com/ZcashFoundation/zebra/pull/5240))
- doc(ci): Fix incorrect docs about Docker build cache order ([#5190](https://github.com/ZcashFoundation/zebra/pull/5190))
- doc(ci): List temporary CI errors that should be retried ([#5281](https://github.com/ZcashFoundation/zebra/pull/5281))
- doc(release): Update release template for Docker releases ([#5193](https://github.com/ZcashFoundation/zebra/pull/5193))
- fix(build): Prefer the PR cache for docker images ([#5282](https://github.com/ZcashFoundation/zebra/pull/5282))
- fix(ci): Comment out empty major version list in release drafter ([#5278](https://github.com/ZcashFoundation/zebra/pull/5278))
- fix(ci): Delete Google Cloud test instances after 3 days ([#5192](https://github.com/ZcashFoundation/zebra/pull/5192))
- fix(ci): Reduce false positives for the `C-trivial` label ([#5309](https://github.com/ZcashFoundation/zebra/pull/5309))
- fix(ci): Revert "bump tj-actions/changed-files from 29.0.7 to 31.0.1" to fix workflow configuration errors ([#5264](https://github.com/ZcashFoundation/zebra/pull/5264))
- fix(clippy): Allow a clippy::bool\_to\_int\_with\_if ([#5300](https://github.com/ZcashFoundation/zebra/pull/5300))
- fix(clippy): actually fix clippy::bool\_to\_int\_with\_if ([#5350](https://github.com/ZcashFoundation/zebra/pull/5350))
- fix(docker): Add edge tag to Docker images, document when latest tag will appear ([#5312](https://github.com/ZcashFoundation/zebra/pull/5312))
- fix(release): Fix the config for `release-drafter` ([#5275](https://github.com/ZcashFoundation/zebra/pull/5275))
- fix(test): Add a timeout to the non\_blocking\_logger test ([#5325](https://github.com/ZcashFoundation/zebra/pull/5325))
- fix(test): Allow more time between thread CPU slices in db\_init\_outside\_future\_executor ([#5310](https://github.com/ZcashFoundation/zebra/pull/5310))
- fix(test): Wait for zebrad and lightwalletd to reach the tip in tests, to improve test coverage ([#5164](https://github.com/ZcashFoundation/zebra/pull/5164))
- fix(test): resolve a timing issue in mempool test ([#5322](https://github.com/ZcashFoundation/zebra/pull/5322))
- fix(test): resolve a timing issue in mempool test ([#5347](https://github.com/ZcashFoundation/zebra/pull/5347))

**Comment by teor2345** [comment]
We have the changelog and version bumps, so I think we're ready for review here

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,55 @@ All notable changes to Zebra are documented in this file.
...
+- Make default command work in docker images, disable optional listener ports ([#5313](https://github.com/ZcashFoundation/zebra/pull/5313))
+- Update yanked versions of cpufeatures (orchard/aes), ed25519 (tor), and quick-xml (flamegraph) ([#5308](https://github.com/ZcashFoundation/zebra/pull/5308))
+- Bump rand\_core from 0.6.3 to 0.6.4, fixes unsoundness bug ([#5175](https://github.com/ZcashFoundation/zebra/pull/5175))
+
+### Changed
+
+- Log git metadata and platform info when zebrad starts up ([#5200](https://github.com/ZcashFoundation/zebra/pull/5200))
+- Update mainnet and testnet checkpoints ([#5360](https://github.com/ZcashFoundation/zebra/pull/5360))
+- Update README for the release candidate ([#5314](https://github.com/ZcashFoundation/zebra/pull/5314))
+- change(docs): Add links to CI/CD docs in the zebra book's sidebar ([#5355](https://github.com/ZcashFoundation/zebra/pull/5355))
```
Comment (arya2) [inline CHANGELOG.md:37]: #5360, #5314, and #5355 are listed as removed in the [removed entries comment](https://github.com/ZcashFoundation/zebra/pull/5383#issuecomment-1275345568)

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,55 @@ All notable changes to Zebra are documented in this file.
 
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/), and this project adheres to [Semantic Versioning](https://semver.org).
 
+## [Zebra 1.0.0-rc.0](https://github.com/ZcashFoundation/zebra/releases/tag/v1.0.0-rc.0) - 2022-10-12
+
+This is the first Zebra release candidate. Zebra's consensus rules, node sync, and `lightwalletd` RPCs are ready for final testing and experimental use. Zebra has not been audited yet.
```
Comment (teor2345) [inline CHANGELOG.md]: Is this how we want to explain the release candidate status?

I think we wrote it down somewhere, but I'm not sure where to find it.

File: CHANGELOG.md (hunk 1 above)
Comment (teor2345) [reply CHANGELOG.md:37]: Thanks, there were some duplicates, I've removed them from the comment.

File: CHANGELOG.md (hunk 2 above)
Comment (teor2345) [reply CHANGELOG.md]:
```suggestion
This is the first Zebra release candidate. Zebra's consensus rules, node sync, and `lightwalletd` RPCs are ready for user testing and experimental use. Zebra has not been audited yet.
```

**Review by teor2345 — DISMISSED** [review]
I'm happy with the version bumps, but I need someone to review the changelog.

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,6 +4,55 @@ All notable changes to Zebra are documented in this file.
...
+
+This release improves RPC performance:
+- Initial `lightwalletd` sync is about twice as fast ([#5307](https://github.com/ZcashFoundation/zebra/pull/5307))
+- RPCs can run while a block is being committed to the state, previously they could be delayed by 5-15 seconds ([#5134](https://github.com/ZcashFoundation/zebra/pull/5134), [#5257](https://github.com/ZcashFoundation/zebra/pull/5257))
+
+### Security
+
+- Make default command work in docker images, disable optional listener ports ([#5313](https://github.com/ZcashFoundation/zebra/pull/5313))
+- Update yanked versions of cpufeatures (orchard/aes), ed25519 (tor), and quick-xml (flamegraph) ([#5308](https://github.com/ZcashFoundation/zebra/pull/5308))
+- Bump rand\_core from 0.6.3 to 0.6.4, fixes unsoundness bug ([#5175](https://github.com/ZcashFoundation/zebra/pull/5175))
```
Comment (arya2) [inline CHANGELOG.md:30]:
```suggestion
- Bump rand_core from 0.6.3 to 0.6.4, fixes unsoundness bug ([#5175](https://github.com/ZcashFoundation/zebra/pull/5175))
```

**Review by arya2 — COMMENTED** [review]
Aside from the small typo and perhaps how we want to explain the release candidate status, the changelog looks good to me.

File: CHANGELOG.md (hunk 3 above)
Comment (teor2345) [reply CHANGELOG.md:30]: The escape avoids activating italics if there are multiple underscores.

This renders ok in Markdown:
Screen Shot 2022-10-12 at 10 53 11

File: CHANGELOG.md (hunk 2 above)
Comment (teor2345) [reply CHANGELOG.md]: I'm not going to block the release on this, we can edit the exact wording later if we want to.

**Review by teor2345 — APPROVED** [review]
Let's go!

**Comment by teor2345** [comment]
@arya2 feel free to turn off merge freeze once you've tested the docker image.

(I'm not in front of a device that can do that right now.)

**Pull request merged**